### DISCUSSION TITLE
Allow full STATIC_SERVER_BASE if scheme provided

### DISCUSSION THREAD

**Discussion by kickin6** [mrhan1993/FooocusWithFastAPI#35] [Ideas]
My use case is I run a container with FooocusAPI with a celery worker that feeds it jobs synchronously one at a time. I spin up multiple of these containers to handle large image generation jobs. They all share a network storage for models and output images. The public side image retrieval is handled by either Nginx or a frontend API.

Being able to provide FooocusAPI the public image URL is important because the worker pods are short-lived and will go away so their API won't be able to serve the image.

Would you consider some logic that if the base-url has a scheme, to use the entire string?

I would spin up FooocusAPI something like `python launch.py --base-url=http://example.com`

Reference code below and my proposal at the bottom.

Cheers

https://github.com/mrhan1993/FooocusAPI/blob/e9d2485a92a0d253379fbbf4dd977da8edda63ff/apis/api.py#L41-L59

```
from urllib.parse import urlparse

def run_server(arguments):
    """
    Run the FastAPI server
    :param arguments: command line arguments
    """
    if arguments.apikey != "":
        api_utils.APIKEY_AUTH = arguments.apikey

    os.environ["WEBHOOK_URL"] = arguments.webhook_url
    try:
        api_port = int(os.environ['API_PORT'])
    except KeyError:
        try:
            api_port = int(arguments.port) + 1
        except TypeError:
            api_port = int(os.environ["GRADIO_SERVER_PORT"]) + 1

    # Parse the base_url to handle cases where it includes a scheme
    parsed_url = urlparse(arguments.base_url)
    
    if parsed_url.scheme:
        # If a scheme is provided, use the full URL
        file_utils.STATIC_SERVER_BASE = f"{arguments.base_url}"
    else:
        file_utils.STATIC_SERVER_BASE = f"http://{arguments.base_url}:{api_port}"

    uvicorn.run(app, host=arguments.listen, port=api_port)
```

**Comment by mrhan1993** [comment]
thank you, that's nice
